File: UdpClientFactor/src/main/java/UdpChatClient/command/HelpHandler.java
package UdpChatClient.command;

import UdpChatClient.ClientState;
import UdpChatClient.CommandProcessor;
import UdpChatClient.Constants;
import UdpChatClient.HandshakeManager;

public class HelpHandler implements CommandHandler {
    private final CommandProcessor commandProcessor;
    
    public HelpHandler(CommandProcessor commandProcessor) {
        this.commandProcessor = commandProcessor;
    }

    @Override
    public void handle(String args, ClientState clientState, HandshakeManager handshakeManager) {
        commandProcessor.showHelp();
    }

    @Override
    public String getDescription() {
        return Constants.CMD_HELP_DESC;
    }
}

File: lib/ClientUdp/command/help_handler.dart
import 'command_handler.dart';
import '../client_state.dart';
import '../command_processor.dart';
import '../constants.dart';
import '../handshake_manager.dart';

class HelpHandler implements CommandHandler {
  final CommandProcessor commandProcessor;
  
  HelpHandler(this.commandProcessor);

  @override
  void handle(String args, ClientState clientState, HandshakeManager handshakeManager) {
    commandProcessor.showHelp();
  }

  @override
  String getDescription() {
    return Constants.CMD_HELP_DESC;
  }
}
